sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/SortBasedAggregationIterator.scala

```diff
@@ -101,18 +117,16 @@ class SortBasedAggregationIterator(
 
     // The search will stop when we see the next group or there is no
     // input row left in the iter.
-    var hasNext = inputKVIterator.next()
-    while (!findNextPartition && hasNext) {
+    while (!findNextPartition && inputIterator.hasNext) {
       // Get the grouping key.
-      val groupingKey = inputKVIterator.getKey
-      val currentRow = inputKVIterator.getValue
+      val inputRow = inputIterator.next()
+      val groupingKey = groupingKeyProjection(inputRow).copy()
+      val currentRow = inputRow.copy()
```

Comment:
Good catch. We only need these copies in `initialize` since we're buffering the result there. I'll fix this shortly.